components/callbacks.py

```diff
@@ -378,9 +404,149 @@ def greet_new_chat_members(update: Update, context: CallbackContext) -> None:
+    )
+
+    if reply_to:
+        try_to_delete(message)
+
+
+def say_potato_job(context: CallbackContext) -> None:
+    user_id, message, ban_person = cast(Tuple[int, Message, User], cast(Job, context.job).context)
+    context.bot.ban_chat_member(chat_id=ONTOPIC_CHAT_ID, user_id=user_id)
+    context.bot.ban_chat_member(chat_id=OFFTOPIC_CHAT_ID, user_id=user_id)
+
+    text = (
+        "You have been banned for userbot-like behavior. If you are not a userbot and wish to be"
+        f"unbanned, please contact {ban_person.mention_html()}."
+    )
+    message.edit_text(text=text)
+
+
+def say_potato_button(update: Update, context: CallbackContext) -> None:
+    callback_query = cast(CallbackQuery, update.callback_query)
+    _, user_id, correct = cast(str, callback_query.data).split()
+
+    if callback_query.from_user.id != user_id:
+        callback_query.answer(
+            text="This button is obviously not meant for you. 😉", show_alert=True
+        )
+        return
+
+    jobs = cast(JobQueue, context.job_queue).get_jobs_by_name(f"POTATO {user_id}")
+    if not jobs:
+        return
+    job = jobs[0]
+
+    if correct == "True":
+        callback_query.answer(
+            text="Thanks for the verification! Have fun in the group 🙂", show_alert=True
+        )
+    else:
+        callback_query.answer(text="That was wrong. Ciao! 👋", show_alert=True)
+        job.run(context.dispatcher)
+
+    job.schedule_removal()
+
+
+def say_potato_command(update: Update, context: CallbackContext) -> None:
+    message = cast(Message, update.effective_message)
+    message.delete()
+
+    if not message.reply_to_message:
+        return
+
+    user = cast(User, message.reply_to_message.from_user)
+
+    if context.args:
+        try:
+            time_limit = int(context.args[0])
+        except ValueError:
+            time_limit = 60
+    else:
+        time_limit = 60
+
+    correct, incorrect_1, incorrect_2 = random.sample(VEGETABLES, 3)
+
+    message_text = (
+        f"You display behavior that is common for userbots, i.e. automated Telegram "
+        f"accounts that usually produce spam. Please verify that you are not a userbot by "
+        f"clicking the button that says <code>{correct}</code>. If you don't press the button "
+        f"within {time_limit} minutes, you will be banned from the PTB groups. If you miss the "
+        f"time limit but are not a userbot and want to get unbanned, please contact "
+        f"{cast(User, message.from_user).mention_html()}."
+    )
+
+    answers = random.sample([(correct, True), (incorrect_1, False), (incorrect_2, False)], 3)
+    keyboard = InlineKeyboardMarkup.from_row(
+        [
+            InlineKeyboardButton(text=veg, callback_data=f"POTATO {user.id} {truth}")
+            for veg, truth in answers
+        ]
+    )
+
+    potato_message = message.reply_to_message.reply_text(message_text, reply_markup=keyboard)
+    user_id = cast(User, message.reply_to_message.from_user).id
```

Comment:
we can just use `user` from above.
```suggestion
    user_id = user.id
```